cmd/api/src/api/v2/cypher_search.go

```diff
@@ -34,27 +36,68 @@ type CypherSearch struct {
 
 func (s Resources) CypherSearch(response http.ResponseWriter, request *http.Request) {
 	var (
-		payload CypherSearch
-		authCtx = ctx.FromRequest(request).AuthCtx
+		payload       CypherSearch
+		authCtx       = ctx.FromRequest(request).AuthCtx
+		auditLogEntry model.AuditEntry
+		preparedQuery queries.PreparedQuery
+		err           error
 	)
 
 	if err := api.ReadJSONRequestPayloadLimited(&payload, request); err != nil {
 		api.WriteErrorResponse(
 			request.Context(),
 			api.BuildErrorResponse(http.StatusBadRequest, "JSON malformed.", request), response,
 		)
-	} else if preparedQuery, err := s.GraphQuery.PrepareCypherQuery(payload.Query); err != nil {
+		return
+	}
+
+	if preparedQuery, err = s.GraphQuery.PrepareCypherQuery(payload.Query); err != nil {
 		api.WriteErrorResponse(
 			request.Context(),
 			api.BuildErrorResponse(http.StatusBadRequest, err.Error(), request), response,
 		)
-	} else if preparedQuery.HasMutation && !s.Authorizer.AllowsPermission(authCtx, auth.Permissions().GraphDBMutate) {
-		s.Authorizer.AuditLogUnauthorizedAccess(request)
-		api.WriteErrorResponse(
-			request.Context(),
-			api.BuildErrorResponse(http.StatusForbidden, "Permission denied: User may not modify the graph.", request), response,
-		)
-	} else if graphResponse, err := s.GraphQuery.RawCypherSearch(request.Context(), preparedQuery, payload.IncludeProperties); err != nil {
+		return
+	}
+
+	if preparedQuery.HasMutation {
+		if !s.Authorizer.AllowsPermission(authCtx, auth.Permissions().GraphDBMutate) {
+			s.Authorizer.AuditLogUnauthorizedAccess(request)
+			api.WriteErrorResponse(
+				request.Context(),
+				api.BuildErrorResponse(http.StatusForbidden, "Permission denied: User may not modify the graph.", request), response,
+			)
+			return
+		}
+
+		// All mutation attempts must be audit logged even when failed
+		if auditLogEntry, err = model.NewAuditEntry(
+			model.AuditLogActionMutateGraph,
+			model.AuditLogStatusIntent,
+			model.AuditData{"query": preparedQuery.StrippedQuery}); err != nil {
+			api.WriteErrorResponse(
+				request.Context(),
+				api.BuildErrorResponse(http.StatusInternalServerError, api.ErrorResponseDetailsInternalServerError, request),
+				response,
+			)
+			return
+		}
```

Comment:
This block is really difficult to read. Would you mind adjusting the indentation to make it more clear where the error handling code is vs the function call?